src/bcf/report/create_report.rs

```diff
@@ -0,0 +1,295 @@
+                let fasta_length = get_fasta_length(fasta_path);
+
+                if variant.pos() < 75 {
+                    let content = create_report_data(
+                        fasta_path,
+                        var.clone(),
+                        bam_path,
+                        chrom.clone(),
+                        0,
+                        end_position as u64 + 75,
+                    );
+                    visualization = manipulate_json(content, 0, end_position as u64 + 75);
+                } else if variant.pos() + 75 >= fasta_length as i64 {
+                    let content = create_report_data(
+                        fasta_path,
+                        var.clone(),
+                        bam_path,
+                        chrom.clone(),
+                        variant.pos() as u64 - 75,
+                        fasta_length - 1,
+                    );
+                    visualization =
+                        manipulate_json(content, variant.pos() as u64 - 75, fasta_length - 1);
+                } else {
+                    let content = create_report_data(
+                        fasta_path,
+                        var.clone(),
+                        bam_path,
+                        chrom.clone(),
+                        variant.pos() as u64 - 75,
+                        end_position as u64 + 75,
+                    );
+                    visualization = manipulate_json(
+                        content,
+                        variant.pos() as u64 - 75,
+                        end_position as u64 + 75,
+                    );
+                }
+
+                let r = Report {
+                    id: id.clone(),
+                    name: name.clone(),
+                    position: variant.pos(),
+                    reference: rfrce.clone(),
+                    var_type: var.var_type,
+                    alternatives: var.alternatives,
+                    ann: Some(ann_strings.clone()),
+                    vis: visualization.to_string(),
+                };
+
+                reports.push(r);
+            }
+        }
+    }
+
+    Ok(reports.clone())
+}
+
+fn create_report_data(
+    fasta_path: &Path,
+    variant: StaticVariant,
+    bam_path: &Path,
+    chrom: String,
+    from: u64,
+    to: u64,
+) -> Json {
+    let mut data = Vec::new();
+
+    for f in read_fasta(fasta_path.clone(), chrom.clone(), from, to) {
+        let nucleobase = json!(f);
+        data.push(nucleobase);
+    }
+
+    let (bases, matches) = get_static_reads(bam_path, fasta_path, chrom.clone(), from, to);
+
+    for b in bases {
+        let base = json!(b);
+        data.push(base);
+    }
+
+    for m in matches {
+        let mat = json!(m);
+        data.push(mat);
+    }
+
+    data.push(json!(variant));
+
+    let values = Json::from_str(&json!(data).to_string()).unwrap();
+
+    values
+}
+
+fn manipulate_json(data: Json, from: u64, to: u64) -> Value {
```

Comment:
Please add a doc comment explaining what this function does.

Reply:
Added a doc comment with b8be9607f541d707c685bd57391ea7ceac522231 explaining the purpose of the function.